src/PackageSelection/PackageSelection.php

```diff
@@ -577,15 +584,17 @@ private function matchAddr($addr1, $addr2, $len = 0, $chunklen = 32): bool
     }
 
     /**
-     * @param RepositoryInterface[] $repositories
+     * @param array<RepositoryInterface|ConfigurableRepositoryInterface> $repositories
```

Comment:
It should be a union type. I don't see any class that implement `ConfigurableRepositoryInterface` without `RepositoryInterface`.

```suggestion
     * @param array<RepositoryInterface&ConfigurableRepositoryInterface> $repositories
```

Reply:
I don't know if we want to change this from union to intersection types, when I do that there are several branches that would always evaluate to true or become unreachable, and some methods that use one of those for a parameter type start throwing errors as well.

Reply:
Ok then, feel free to mark as resolved.